pkg/storage/bloom/v1/fuse.go

```diff
@@ -142,3 +144,43 @@ func (fq *FusedQuerier) Run() error {
 
 	return nil
 }
+
+// boundedRequests is a set of requests that are clamped to a specific range
+type boundedRequests struct {
+	bounds FingerprintBounds
+	reqs   [][]model.Fingerprint
+}
+
+// reqs models a set of requests covering many fingerprints.
+// consumers models a set of blocks covering different fingerprint ranges
+func partitionFingerprintRange(reqs [][]model.Fingerprint, blocks []FingerprintBounds) (res []boundedRequests) {
```

Comment:
You have a named return value, but you return the value also explicitly.
Personally I'm not a big fan of named return values, because they can hide allocation problems.